.circleci/config.yml

```diff
@@ -0,0 +1,64 @@
+version: 2
+jobs:
+  install:
+    parameters:
+      python-version:
+        type: string
+    working_directory: ~/project
+    machine:
+      image: ubuntu-1604:201903-01
+    steps:
+      - run:
+          name: Set python << parameters.python-version >>
+          command: |
+            pyenv global << parameters.python-version >>
+            python -m pip install --upgrade pip setuptools wheel
+      - checkout
+      # Download and cache dependencies
+      - restore_cache:
+          keys:
+            - machine-<< parameters.python-version >>-{{ checksum "dev_requirements.txt" }}
```

Comment:
2 things here IMO:

- dev_requirements.txt contains requirements that are in the setup.py. It should just contain additional requirements and `-e .` to really test as a user would
- you should also add the setup.py checksum as it is part of the requirements

Eventually, you can also get rid of the dev_requirements by defining some `extra_requires` in the setup.py : https://github.com/sicara/tf-explain/blob/39616c4f2820e04121887ddf6f8d185b43ac63f9/setup.py#L20